bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -545,23 +548,36 @@ public int readSize()
 
   }
 
-  class Subscriber extends AbstractLengthPrependerClient
+  private class Subscriber extends WriteOnlyLengthPrependerClient
   {
     private LogicalNode ln;
 
     Subscriber(LogicalNode ln, int bufferSize)
     {
-      super(1024, bufferSize);
+      super(1024 * 1024, bufferSize == 0 ? 256 * 1024 : bufferSize);
       this.ln = ln;
       ln.addConnection(this);
-      super.write = false;
     }
 
     @Override
-    public void onMessage(byte[] buffer, int offset, int size)
+    public void connected()
     {
-      logger.warn("Received data when no data is expected: {}",
-          Arrays.toString(Arrays.copyOfRange(buffer, offset, offset + size)));
+      super.connected();
+      serverHelperExecutor.submit(new Runnable()
```

Comment:
Should the subscriber and publisher be handled in a separate executor or do we need to add more threads to the serverHelperExecutor? As serverHelperExecutor is also used during datalist flush, which may delay the subscription.